site/src/Map.jsx

```diff
@@ -1,135 +1,313 @@
+  latitude: 51.05,
   longitude: 3.7303,
   zoom: 16,
   bearing: 0,
-  pitch: 0
+  pitch: 0,
+};
+
+// this reference must remain constant to avoid re-renders
+const MAP_STYLE = {
+  version: 8,
+  glyphs: "https://demotiles.maplibre.org/font/{fontstack}/{range}.pbf",
+  sources: {},
+  layers: [],
+};
+
+const ThemeSource = ({ name, url }) => {
+  return <Source id={name} type="vector" url={`${url}${name}.pmtiles`} />;
+};
+
+ThemeSource.propTypes = {
+  name: PropTypes.string.isRequired,
+  url: PropTypes.string.isRequired,
+};
+
+const ThemeTypeLayer = ({
+  theme,
+  type,
+  color,
+  point,
+  line,
+  polygon,
+  extrusion,
+}) => {
+  return (
+    <>
+      {point ? (
+        <Layer
+          filter={["==", ["geometry-type"], "Point"]}
+          beforeId="divisions_division"
+          id={`${theme}_${type}_point`}
+          type="circle"
+          source={theme}
+          source-layer={type}
+          paint={{
+            "circle-color": color,
+            "circle-radius": [
+              "interpolate",
+              ["exponential", 2],
+              ["zoom"],
+              0,
+              1,
+              17,
+              8,
+            ],
+          }}
+        />
+      ) : null}
+      {line ? (
+        <Layer
+          filter={["==", ["geometry-type"], "LineString"]}
+          beforeId="divisions_division"
+          id={`${theme}_${type}_line`}
+          type="line"
+          source={theme}
+          source-layer={type}
+          paint={{ "line-color": color }}
+        />
+      ) : null}
+      {polygon ? (
+        <Layer
+          filter={["==", ["geometry-type"], "Polygon"]}
+          beforeId="divisions_division"
+          id={`${theme}_${type}_fill`}
+          type="fill"
+          source={theme}
+          source-layer={type}
+          paint={{ "fill-color": color, "fill-opacity": 0.2 }}
+        />
+      ) : null}
+      {extrusion ? (
+        <Layer
+          filter={[
+            "all",
+            ["==", ["geometry-type"], "Polygon"],
+            ["!=", ["get", "has_parts"], true],
+          ]} // prevent z-fighting
+          beforeId="divisions_division"
+          id={`${theme}_${type}_fill-extrusion`}
+          type="fill-extrusion"
+          source={theme}
+          source-layer={type}
+          paint={{
+            "fill-extrusion-color": color,
+            "fill-extrusion-opacity": 0.35,
+            "fill-extrusion-base": ["get", "min_height"],
+            "fill-extrusion-height": ["get", "height"],
+          }}
+        />
+      ) : null}
+    </>
+  );
 };
-export default function Map({mode}) {
 
-  const [pmTilesReady, setPmTilesReady] = useState(false)
-  const [cursor, setCursor] = useState('auto');
+ThemeTypeLayer.propTypes = {
+  theme: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+  point: PropTypes.bool,
+  line: PropTypes.bool,
+  polygon: PropTypes.bool,
+  extrusion: PropTypes.bool,
+};
```

Comment:
This interface looks good for now, but may become restrictive overtime. For example, we may want to dynamically color places based on type (or display some icons based on presets like the iD editor).